Allow to disable the nomad source

Hi, I think it would be useful to to allow disabling the nomad source.

In our case, we want to deploy two instances of nomad-autoscaler, one dealing with allocation scaling and another with node pools autoscaling. We created several policies so each of them only has whats needed (the allocation one has scale over namespaces and the node autoscaler has permissions over nodes only). Right now as they always get the policies from nomad, the one dealing with node pools (that gets the policies from the filesystem) gets and error constantly because it's trying to get them.

```
[ERROR] policy_manager: failed to call the Nomad list policies API: Unexpected response code: 403 (Permission denied)
```

I think allowing to disable the sourcing of policies from nomad would be a nice addition.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 ## UNRELEASED
 
 IMPROVEMENTS:
+ * agent: Allow disabling specific policy sources [[GH-544](https://github.com/hashicorp/nomad-autoscaler/pull/544)]
  * agent: Dispense `fixed-value`, `pass-through`, and `threshold` strategy plugins by default [[GH-536](https://github.com/hashicorp/nomad-autoscaler/pull/536)]
  * policy: Prevent scaling cluster to zero when using the Nomad APM [[GH-534](https://github.com/hashicorp/nomad-autoscaler/pull/534)]
  * scaleutils: Add combined filter to allow filtering by node class and datacenter [[GH-535](https://github.com/hashicorp/nomad-autoscaler/pull/535)]

--- a/agent/agent.go
+++ b/agent/agent.go
@@ -77,7 +77,11 @@ func (a *Agent) Run() error {
 	}
 	a.inMemSink = inMem
 
-	policyEvalCh := a.setupPolicyManager()
+	// Setup policy manager.
+	policyEvalCh, err := a.setupPolicyManager()
+	if err != nil {
+		return fmt.Errorf("failed to setup policy manager: %v", err)
+	}
 	go a.policyManager.Run(ctx, policyEvalCh)
 
 	// Launch eval broker and workers.
@@ -131,7 +135,7 @@ func (a *Agent) initWorkers(ctx context.Context) {
 	}
 }
 
-func (a *Agent) setupPolicyManager() chan *sdk.ScalingEvaluation {
+func (a *Agent) setupPolicyManager() (chan *sdk.ScalingEvaluation, error) {
 
 	// Create our processor, a shared method for performing basic policy
 	// actions.
@@ -142,20 +146,34 @@ func (a *Agent) setupPolicyManager() chan *sdk.ScalingEvaluation {
 	policyProcessor := policy.NewProcessor(&cfgDefaults, a.getNomadAPMNames())
 
 	// Setup our initial default policy source which is Nomad.
-	sources := map[policy.SourceName]policy.Source{
-		policy.SourceNameNomad: nomadPolicy.NewNomadSource(a.logger, a.nomadClient, policyProcessor),
+	sources := map[policy.SourceName]policy.Source{}
+	for _, s := range a.config.PolicySources {
+		if !s.Enabled {
+			continue
+		}
+
+		switch policy.SourceName(s.Name) {
+		case policy.SourceNameNomad:
+			sources[policy.SourceNameNomad] = nomadPolicy.NewNomadSource(a.logger, a.nomadClient, policyProcessor)
+		case policy.SourceNameFile:
+			// Only setup the file source if operators have configured a
+			// scaling policy directory to read from.
+			if a.config.Policy.Dir != "" {
+				sources[policy.SourceNameFile] = filePolicy.NewFileSource(a.logger, a.config.Policy.Dir, policyProcessor)
+			}
+		}
 	}
 
-	// If the operators has configured a scaling policy directory to read from
-	// then setup the file source.
-	if a.config.Policy.Dir != "" {
-		sources[policy.SourceNameFile] = filePolicy.NewFileSource(a.logger, a.config.Policy.Dir, policyProcessor)
+	// TODO: Once full policy source reload is implemented this should probably
+	// be just a warning.
+	if len(sources) == 0 {
+		return nil, fmt.Errorf("no policy source available")
 	}
 
 	a.policySources = sources
 	a.policyManager = policy.NewManager(a.logger, a.policySources, a.pluginManager, a.config.Telemetry.CollectionInterval)
 
-	return make(chan *sdk.ScalingEvaluation, 10)
+	return make(chan *sdk.ScalingEvaluation, 10), nil
 }
 
 func (a *Agent) stop() {

--- a/agent/agent.go
+++ b/agent/agent.go
@@ -148,7 +148,7 @@ func (a *Agent) setupPolicyManager() (chan *sdk.ScalingEvaluation, error) {
 	// Setup our initial default policy source which is Nomad.
 	sources := map[policy.SourceName]policy.Source{}
 	for _, s := range a.config.Policy.Sources {
-		if !s.Enabled {
+		if s.Enabled == nil || !*s.Enabled {
 			continue
 		}
 

--- a/agent/agent.go
+++ b/agent/agent.go
@@ -147,7 +147,7 @@ func (a *Agent) setupPolicyManager() (chan *sdk.ScalingEvaluation, error) {
 
 	// Setup our initial default policy source which is Nomad.
 	sources := map[policy.SourceName]policy.Source{}
-	for _, s := range a.config.PolicySources {
+	for _, s := range a.config.Policy.Sources {
 		if !s.Enabled {
 			continue
 		}

--- a/agent/config/config.go
+++ b/agent/config/config.go
@@ -65,10 +65,9 @@ type Agent struct {
 	// Consul is used to configure a consul API client
 	Consul *Consul `hcl:"consul,block"`
 
-	PolicySources []*PolicySource `hcl:"source,block"`
-	APMs          []*Plugin       `hcl:"apm,block"`
-	Targets       []*Plugin       `hcl:"target,block"`
-	Strategies    []*Plugin       `hcl:"strategy,block"`
+	APMs       []*Plugin `hcl:"apm,block"`
+	Targets    []*Plugin `hcl:"target,block"`
+	Strategies []*Plugin `hcl:"strategy,block"`
 }
 
 // DynamicApplicationSizing contains configuration values to control the
@@ -458,6 +457,9 @@ type Policy struct {
 	// `evaluation_interval` is not defined in a policy.
 	DefaultEvaluationInterval    time.Duration
 	DefaultEvaluationIntervalHCL string `hcl:"default_evaluation_interval,optional" json:"-"`
+
+	// Sources store configuration for policy sources.
+	Sources []*PolicySource `hcl:"source,block"`
 }
 
 // PolicyEval holds the configuration related to the policy evaluation process.
@@ -560,15 +562,15 @@ func Default() (*Agent, error) {
 		Policy: &Policy{
 			DefaultCooldown:           defaultPolicyCooldown,
 			DefaultEvaluationInterval: defaultEvaluationInterval,
+			Sources: []*PolicySource{
+				{Name: policySourceFile, Enabled: true},
+				{Name: policySourceNomad, Enabled: true},
+			},
 		},
 		PolicyEval: &PolicyEval{
 			DeliveryLimit: defaultPolicyEvalDeliveryLimit,
 			AckTimeout:    defaultPolicyEvalAckTimeout,
 			Workers:       defaultPolicyEvalWorkers,
-		},
-		PolicySources: []*PolicySource{
-			{Name: policySourceFile, Enabled: true},
-			{Name: policySourceNomad, Enabled: true},
 		},
 		APMs: []*Plugin{
 			{Name: plugins.InternalAPMNomad, Driver: plugins.InternalAPMNomad},
@@ -634,16 +636,6 @@ func (a *Agent) Merge(b *Agent) *Agent {
 		result.PolicyEval = result.PolicyEval.merge(b.PolicyEval)
 	}
 
-	if len(result.PolicySources) == 0 && len(b.PolicySources) != 0 {
-		sourceCopy := make([]*PolicySource, len(b.PolicySources))
-		for i, v := range b.PolicySources {
-			sourceCopy[i] = v.copy()
-		}
-		result.PolicySources = sourceCopy
-	} else if len(b.PolicySources) != 0 {
-		result.PolicySources = policySourceConfigSetMerge(result.PolicySources, b.PolicySources)
-	}
-
 	if len(result.APMs) == 0 && len(b.APMs) != 0 {
 		apmCopy := make([]*Plugin, len(b.APMs))
 		for i, v := range b.APMs {
@@ -687,8 +679,10 @@ func (a *Agent) Validate() error {
 		result = multierror.Append(result, a.PolicyEval.validate())
 	}
 
-	for _, s := range a.PolicySources {
-		result = multierror.Append(result, s.validate())
+	if a.Policy != nil {
+		for _, s := range a.Policy.Sources {
+			result = multierror.Append(result, s.validate())
+		}
 	}
 
 	return result.ErrorOrNil()
@@ -914,6 +908,17 @@ func (p *Policy) merge(b *Policy) *Policy {
 	if b.DefaultEvaluationInterval != 0 {
 		result.DefaultEvaluationInterval = b.DefaultEvaluationInterval
 	}
+
+	if len(result.Sources) == 0 && len(b.Sources) != 0 {
+		sourceCopy := make([]*PolicySource, len(b.Sources))
+		for i, v := range b.Sources {
+			sourceCopy[i] = v.copy()
+		}
+		result.Sources = sourceCopy
+	} else if len(b.Sources) != 0 {
+		result.Sources = policySourceConfigSetMerge(result.Sources, b.Sources)
+	}
+
 	return &result
 }
 
@@ -1110,6 +1115,15 @@ func parseFile(file string, cfg *Agent) error {
 			}
 			cfg.Policy.DefaultEvaluationInterval = d
 		}
+
+		for _, source := range cfg.Policy.Sources {
+			if source.EnabledPtr != nil {
+				source.Enabled = *source.EnabledPtr
+			} else {
+				// Default to true if source block is defined.
+				source.Enabled = true
+			}
+		}
 	}
 
 	if cfg.Telemetry != nil {
@@ -1161,14 +1175,6 @@ func parseFile(file string, cfg *Agent) error {
 		}
 	}
 
-	for _, source := range cfg.PolicySources {
-		if source.EnabledPtr != nil {
-			source.Enabled = *source.EnabledPtr
-		} else {
-			// Default to true if source block is defined.
-			source.Enabled = true
-		}
-	}
 	return nil
 }
 

--- a/agent/config/config.go
+++ b/agent/config/config.go
@@ -65,9 +65,10 @@ type Agent struct {
 	// Consul is used to configure a consul API client
 	Consul *Consul `hcl:"consul,block"`
 
-	APMs       []*Plugin `hcl:"apm,block"`
-	Targets    []*Plugin `hcl:"target,block"`
-	Strategies []*Plugin `hcl:"strategy,block"`
+	PolicySources []*PolicySource `hcl:"source,block"`
+	APMs          []*Plugin       `hcl:"apm,block"`
+	Targets       []*Plugin       `hcl:"target,block"`
+	Strategies    []*Plugin       `hcl:"strategy,block"`
 }
 
 // DynamicApplicationSizing contains configuration values to control the
@@ -475,6 +476,13 @@ type PolicyEval struct {
 	Workers map[string]int `hcl:"workers,optional"`
 }
 
+// PolicySource is an individual configured policy source.
+type PolicySource struct {
+	Name       string `hcl:"name,label"`
+	EnabledPtr *bool  `hcl:"enabled,optional"`
+	Enabled    bool
+}
+
 const (
 	// defaultLogLevel is the default log level used for the Autoscaler agent.
 	defaultLogLevel = "info"
@@ -508,6 +516,17 @@ const (
 	// defaultPolicyWorkerAckTimeout is the default time limit that a policy
 	// eval must be ACK'd.
 	defaultPolicyEvalAckTimeout = 5 * time.Minute
+)
+
+// TODO: there's an unexpected import cycle that prevents us from using the
+// values defined in policy/source.go.
+const (
+	// policySourceFile is the source for policies that are loaded from disk.
+	policySourceFile = "file"
+
+	// policySourceNomad is the source for policies that originate from the
+	// Nomad scaling policies API.
+	policySourceNomad = "nomad"
 )
 
 var defaultPolicyEvalWorkers = map[string]int{
@@ -547,14 +566,22 @@ func Default() (*Agent, error) {
 			AckTimeout:    defaultPolicyEvalAckTimeout,
 			Workers:       defaultPolicyEvalWorkers,
 		},
-		APMs: []*Plugin{{Name: plugins.InternalAPMNomad, Driver: plugins.InternalAPMNomad}},
+		PolicySources: []*PolicySource{
+			{Name: policySourceFile, Enabled: true},
+			{Name: policySourceNomad, Enabled: true},
+		},
+		APMs: []*Plugin{
+			{Name: plugins.InternalAPMNomad, Driver: plugins.InternalAPMNomad},
+		},
 		Strategies: []*Plugin{
 			{Name: plugins.InternalStrategyFixedValue, Driver: plugins.InternalStrategyFixedValue},
 			{Name: plugins.InternalStrategyPassThrough, Driver: plugins.InternalStrategyPassThrough},
 			{Name: plugins.InternalStrategyTargetValue, Driver: plugins.InternalStrategyTargetValue},
 			{Name: plugins.InternalStrategyThreshold, Driver: plugins.InternalStrategyThreshold},
 		},
-		Targets: []*Plugin{{Name: plugins.InternalTargetNomad, Driver: plugins.InternalTargetNomad}},
+		Targets: []*Plugin{
+			{Name: plugins.InternalTargetNomad, Driver: plugins.InternalTargetNomad},
+		},
 	}, nil
 }
 
@@ -607,6 +634,16 @@ func (a *Agent) Merge(b *Agent) *Agent {
 		result.PolicyEval = result.PolicyEval.merge(b.PolicyEval)
 	}
 
+	if len(result.PolicySources) == 0 && len(b.PolicySources) != 0 {
+		sourceCopy := make([]*PolicySource, len(b.PolicySources))
+		for i, v := range b.PolicySources {
+			sourceCopy[i] = v.copy()
+		}
+		result.PolicySources = sourceCopy
+	} else if len(b.PolicySources) != 0 {
+		result.PolicySources = policySourceConfigSetMerge(result.PolicySources, b.PolicySources)
+	}
+
 	if len(result.APMs) == 0 && len(b.APMs) != 0 {
 		apmCopy := make([]*Plugin, len(b.APMs))
 		for i, v := range b.APMs {
@@ -648,6 +685,10 @@ func (a *Agent) Validate() error {
 
 	if a.PolicyEval != nil {
 		result = multierror.Append(result, a.PolicyEval.validate())
+	}
+
+	for _, s := range a.PolicySources {
+		result = multierror.Append(result, s.validate())
 	}
 
 	return result.ErrorOrNil()
@@ -921,6 +962,56 @@ func (pw *PolicyEval) validate() *multierror.Error {
 	}
 	return result
 }
+func (s *PolicySource) copy() *PolicySource {
+	if s == nil {
+		return nil
+	}
+
+	return &PolicySource{
+		Name:       s.Name,
+		EnabledPtr: s.EnabledPtr,
+		Enabled:    s.Enabled,
+	}
+}
+
+func (s *PolicySource) merge(b *PolicySource) *PolicySource {
+	if s == nil {
+		return b
+	}
+
+	result := *s
+
+	if len(b.Name) != 0 {
+		result.Name = b.Name
+	}
+	if b.EnabledPtr != nil {
+		result.EnabledPtr = b.EnabledPtr
+		result.Enabled = b.Enabled
+	}
+
+	return &result
+}
+
+func (s *PolicySource) validate() *multierror.Error {
+	var result *multierror.Error
+	prefix := fmt.Sprintf("source[%s] ->", s.Name)
+
+	validSources := map[string]bool{
+		policySourceNomad: true,
+		policySourceFile:  true,
+	}
+	if _, ok := validSources[s.Name]; !ok {
+		result = multierror.Append(result, fmt.Errorf("invalid source %q", s.Name))
+	}
+
+	// Prefix all errors.
+	if result != nil {
+		for i, err := range result.Errors {
+			result.Errors[i] = multierror.Prefix(err, prefix)
+		}
+	}
+	return result
+}
 
 // pluginConfigSetMerge merges two sets of plugin configs. For plugins with the
 // same name, the configs are merged.
@@ -956,6 +1047,43 @@ func pluginConfigSetMerge(first, second []*Plugin) []*Plugin {
 		}
 
 		out = append(out, plugin)
+	}
+
+	return out
+}
+
+func policySourceConfigSetMerge(first, second []*PolicySource) []*PolicySource {
+	findex := make(map[string]*PolicySource, len(first))
+	for _, p := range first {
+		findex[p.Name] = p
+	}
+
+	sindex := make(map[string]*PolicySource, len(second))
+	for _, p := range second {
+		sindex[p.Name] = p
+	}
+
+	var out []*PolicySource
+
+	// Go through the first set and merge any value that exist in both
+	for sourceName, original := range findex {
+		second, ok := sindex[sourceName]
+		if !ok {
+			out = append(out, original.copy())
+			continue
+		}
+
+		out = append(out, original.merge(second))
+	}
+
+	// Go through the second set and add any value that didn't exist in both
+	for sourceName, source := range sindex {
+		_, ok := findex[sourceName]
+		if ok {
+			continue
+		}
+
+		out = append(out, source)
 	}
 
 	return out
@@ -1030,6 +1158,15 @@ func parseFile(file string, cfg *Agent) error {
 				return err
 			}
 			cfg.DynamicApplicationSizing.EvaluateAfter = t
+		}
+	}
+
+	for _, source := range cfg.PolicySources {
+		if source.EnabledPtr != nil {
+			source.Enabled = *source.EnabledPtr
+		} else {
+			// Default to true if source block is defined.
+			source.Enabled = true
 		}
 	}
 	return nil

--- a/agent/config/config.go
+++ b/agent/config/config.go
@@ -15,6 +15,7 @@ import (
 	"github.com/hashicorp/hcl/v2/hclsimple"
 	"github.com/hashicorp/nomad-autoscaler/plugins"
 	"github.com/hashicorp/nomad-autoscaler/sdk/helper/file"
+	"github.com/hashicorp/nomad-autoscaler/sdk/helper/ptr"
 	"github.com/mitchellh/copystructure"
 )
 
@@ -480,9 +481,8 @@ type PolicyEval struct {
 
 // PolicySource is an individual configured policy source.
 type PolicySource struct {
-	Name       string `hcl:"name,label"`
-	EnabledPtr *bool  `hcl:"enabled,optional"`
-	Enabled    bool
+	Name    string `hcl:"name,label"`
+	Enabled *bool  `hcl:"enabled,optional"`
 }
 
 const (
@@ -563,8 +563,8 @@ func Default() (*Agent, error) {
 			DefaultCooldown:           defaultPolicyCooldown,
 			DefaultEvaluationInterval: defaultEvaluationInterval,
 			Sources: []*PolicySource{
-				{Name: policySourceFile, Enabled: true},
-				{Name: policySourceNomad, Enabled: true},
+				{Name: policySourceFile, Enabled: ptr.BoolToPtr(true)},
+				{Name: policySourceNomad, Enabled: ptr.BoolToPtr(true)},
 			},
 		},
 		PolicyEval: &PolicyEval{
@@ -972,10 +972,14 @@ func (s *PolicySource) copy() *PolicySource {
 		return nil
 	}
 
+	var enabled *bool
+	if s.Enabled != nil {
+		enabled = ptr.BoolToPtr(*s.Enabled)
+	}
+
 	return &PolicySource{
-		Name:       s.Name,
-		EnabledPtr: s.EnabledPtr,
-		Enabled:    s.Enabled,
+		Name:    s.Name,
+		Enabled: enabled,
 	}
 }
 
@@ -989,8 +993,7 @@ func (s *PolicySource) merge(b *PolicySource) *PolicySource {
 	if len(b.Name) != 0 {
 		result.Name = b.Name
 	}
-	if b.EnabledPtr != nil {
-		result.EnabledPtr = b.EnabledPtr
+	if b.Enabled != nil {
 		result.Enabled = b.Enabled
 	}
 
@@ -1117,11 +1120,9 @@ func parseFile(file string, cfg *Agent) error {
 		}
 
 		for _, source := range cfg.Policy.Sources {
-			if source.EnabledPtr != nil {
-				source.Enabled = *source.EnabledPtr
-			} else {
+			if source.Enabled == nil {
 				// Default to true if source block is defined.
-				source.Enabled = true
+				source.Enabled = ptr.BoolToPtr(true)
 			}
 		}
 	}

--- a/agent/config/config_test.go
+++ b/agent/config/config_test.go
@@ -24,10 +24,10 @@ func Test_Default(t *testing.T) {
 	assert.Equal(t, "127.0.0.1", def.HTTP.BindAddress)
 	assert.Equal(t, 8080, def.HTTP.BindPort)
 	assert.Equal(t, def.Policy.DefaultCooldown, 5*time.Minute)
+	assert.Len(t, def.Policy.Sources, 2)
 	assert.Equal(t, defaultPolicyEvalDeliveryLimit, def.PolicyEval.DeliveryLimit)
 	assert.Equal(t, defaultPolicyEvalAckTimeout, def.PolicyEval.AckTimeout)
 	assert.Equal(t, defaultPolicyEvalWorkers, def.PolicyEval.Workers)
-	assert.Len(t, def.PolicySources, 2)
 	assert.Len(t, def.APMs, 1)
 	assert.Len(t, def.Targets, 1)
 	assert.Len(t, def.Strategies, 4)
@@ -56,11 +56,13 @@ func TestAgent_Merge(t *testing.T) {
 				"some-other": 3,
 			},
 		},
-		PolicySources: []*PolicySource{
-			{
-				Name:       "nomad",
-				EnabledPtr: ptr.BoolToPtr(false),
-				Enabled:    false,
+		Policy: &Policy{
+			Sources: []*PolicySource{
+				{
+					Name:       "nomad",
+					EnabledPtr: ptr.BoolToPtr(false),
+					Enabled:    false,
+				},
 			},
 		},
 		APMs: []*Plugin{
@@ -108,6 +110,18 @@ func TestAgent_Merge(t *testing.T) {
 			Dir:                       "/etc/scaling/policies",
 			DefaultCooldown:           20 * time.Minute,
 			DefaultEvaluationInterval: 10 * time.Second,
+			Sources: []*PolicySource{
+				{
+					Name:       "file",
+					EnabledPtr: ptr.BoolToPtr(false),
+					Enabled:    false,
+				},
+				{
+					Name:       "nomad",
+					EnabledPtr: ptr.BoolToPtr(true),
+					Enabled:    true,
+				},
+			},
 		},
 		PolicyEval: &PolicyEval{
 			DeliveryLimitPtr: ptr.IntToPtr(10),
@@ -140,18 +154,6 @@ func TestAgent_Merge(t *testing.T) {
 			CirconusCheckDisplayName:           "some-name",
 			CirconusBrokerID:                   "some-id",
 			CirconusBrokerSelectTag:            "some-other-tag",
-		},
-		PolicySources: []*PolicySource{
-			{
-				Name:       "file",
-				EnabledPtr: ptr.BoolToPtr(false),
-				Enabled:    false,
-			},
-			{
-				Name:       "nomad",
-				EnabledPtr: ptr.BoolToPtr(true),
-				Enabled:    true,
-			},
 		},
 		APMs: []*Plugin{
 			{
@@ -210,6 +212,18 @@ func TestAgent_Merge(t *testing.T) {
 			Dir:                       "/etc/scaling/policies",
 			DefaultCooldown:           20 * time.Minute,
 			DefaultEvaluationInterval: 10 * time.Second,
+			Sources: []*PolicySource{
+				{
+					Name:       "file",
+					EnabledPtr: ptr.BoolToPtr(false),
+					Enabled:    false,
+				},
+				{
+					Name:       "nomad",
+					EnabledPtr: ptr.BoolToPtr(true),
+					Enabled:    true,
+				},
+			},
 		},
 		PolicyEval: &PolicyEval{
 			DeliveryLimitPtr: ptr.IntToPtr(10),
@@ -244,18 +258,6 @@ func TestAgent_Merge(t *testing.T) {
 			CirconusCheckDisplayName:           "some-name",
 			CirconusBrokerID:                   "some-id",
 			CirconusBrokerSelectTag:            "some-other-tag",
-		},
-		PolicySources: []*PolicySource{
-			{
-				Name:       "file",
-				EnabledPtr: ptr.BoolToPtr(false),
-				Enabled:    false,
-			},
-			{
-				Name:       "nomad",
-				EnabledPtr: ptr.BoolToPtr(true),
-				Enabled:    true,
-			},
 		},
 		APMs: []*Plugin{
 			{
@@ -315,7 +317,7 @@ func TestAgent_Merge(t *testing.T) {
 	assert.Equal(t, expectedResult.PluginDir, actualResult.PluginDir)
 	assert.Equal(t, expectedResult.Policy, actualResult.Policy)
 	assert.Equal(t, expectedResult.PolicyEval, actualResult.PolicyEval)
-	assert.ElementsMatch(t, expectedResult.PolicySources, actualResult.PolicySources)
+	assert.ElementsMatch(t, expectedResult.Policy.Sources, actualResult.Policy.Sources)
 	assert.ElementsMatch(t, expectedResult.APMs, actualResult.APMs)
 	assert.ElementsMatch(t, expectedResult.Targets, actualResult.Targets)
 	assert.ElementsMatch(t, expectedResult.Strategies, actualResult.Strategies)
@@ -442,7 +444,10 @@ func TestAgent_policySources(t *testing.T) {
 
 	// Enabled by default if block is present.
 	cfg := &Agent{}
-	nomadSourceCfg := `source "nomad" {}`
+	nomadSourceCfg := `
+policy {
+  source "nomad" {}
+}`
 
 	_, err = fh.WriteString(nomadSourceCfg)
 	require.NoError(t, err)
@@ -455,14 +460,16 @@ func TestAgent_policySources(t *testing.T) {
 			Enabled:    true,
 		},
 	}
-	assert.ElementsMatch(t, expected, cfg.PolicySources)
+	assert.ElementsMatch(t, expected, cfg.Policy.Sources)
 
 	// Disable "nomad" policy source.
 	cfg = &Agent{}
 
 	noNomadSourceCfg := `
-source "nomad" {
-  enabled = false
+policy {
+  source "nomad" {
+    enabled = false
+  }
 }`
 
 	// Reset the test file.
@@ -488,14 +495,16 @@ source "nomad" {
 			Enabled:    false,
 		},
 	}
-	assert.ElementsMatch(t, expected, result.PolicySources)
+	assert.ElementsMatch(t, expected, result.Policy.Sources)
 
 	// Disable "file" policy source.
 	cfg = &Agent{}
 
 	noFileSourceCfg := `
-source "file" {
-  enabled = false
+policy {
+  source "file" {
+    enabled = false
+  }
 }`
 
 	// Reset the test file.
@@ -521,5 +530,5 @@ source "file" {
 			Enabled:    true,
 		},
 	}
-	assert.ElementsMatch(t, expected, result.PolicySources)
+	assert.ElementsMatch(t, expected, result.Policy.Sources)
 }

--- a/agent/config/config_test.go
+++ b/agent/config/config_test.go
@@ -10,6 +10,7 @@ import (
 
 	"github.com/hashicorp/nomad-autoscaler/sdk/helper/ptr"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func Test_Default(t *testing.T) {
@@ -26,6 +27,7 @@ func Test_Default(t *testing.T) {
 	assert.Equal(t, defaultPolicyEvalDeliveryLimit, def.PolicyEval.DeliveryLimit)
 	assert.Equal(t, defaultPolicyEvalAckTimeout, def.PolicyEval.AckTimeout)
 	assert.Equal(t, defaultPolicyEvalWorkers, def.PolicyEval.Workers)
+	assert.Len(t, def.PolicySources, 2)
 	assert.Len(t, def.APMs, 1)
 	assert.Len(t, def.Targets, 1)
 	assert.Len(t, def.Strategies, 4)
@@ -52,6 +54,13 @@ func TestAgent_Merge(t *testing.T) {
 			Workers: map[string]int{
 				"horizontal": 5,
 				"some-other": 3,
+			},
+		},
+		PolicySources: []*PolicySource{
+			{
+				Name:       "nomad",
+				EnabledPtr: ptr.BoolToPtr(false),
+				Enabled:    false,
 			},
 		},
 		APMs: []*Plugin{
@@ -131,6 +140,18 @@ func TestAgent_Merge(t *testing.T) {
 			CirconusCheckDisplayName:           "some-name",
 			CirconusBrokerID:                   "some-id",
 			CirconusBrokerSelectTag:            "some-other-tag",
+		},
+		PolicySources: []*PolicySource{
+			{
+				Name:       "file",
+				EnabledPtr: ptr.BoolToPtr(false),
+				Enabled:    false,
+			},
+			{
+				Name:       "nomad",
+				EnabledPtr: ptr.BoolToPtr(true),
+				Enabled:    true,
+			},
 		},
 		APMs: []*Plugin{
 			{
@@ -224,6 +245,18 @@ func TestAgent_Merge(t *testing.T) {
 			CirconusBrokerID:                   "some-id",
 			CirconusBrokerSelectTag:            "some-other-tag",
 		},
+		PolicySources: []*PolicySource{
+			{
+				Name:       "file",
+				EnabledPtr: ptr.BoolToPtr(false),
+				Enabled:    false,
+			},
+			{
+				Name:       "nomad",
+				EnabledPtr: ptr.BoolToPtr(true),
+				Enabled:    true,
+			},
+		},
 		APMs: []*Plugin{
 			{
 				Name:   "nomad-apm",
@@ -282,6 +315,7 @@ func TestAgent_Merge(t *testing.T) {
 	assert.Equal(t, expectedResult.PluginDir, actualResult.PluginDir)
 	assert.Equal(t, expectedResult.Policy, actualResult.Policy)
 	assert.Equal(t, expectedResult.PolicyEval, actualResult.PolicyEval)
+	assert.ElementsMatch(t, expectedResult.PolicySources, actualResult.PolicySources)
 	assert.ElementsMatch(t, expectedResult.APMs, actualResult.APMs)
 	assert.ElementsMatch(t, expectedResult.Targets, actualResult.Targets)
 	assert.ElementsMatch(t, expectedResult.Strategies, actualResult.Strategies)
@@ -395,4 +429,97 @@ func TestAgent_loadDir(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, "trace", cfg.LogLevel)
 	assert.Equal(t, "/opt/nomad-autoscaler/plugins", cfg.PluginDir)
+}
+
+func TestAgent_policySources(t *testing.T) {
+	defaultConfig, err := Default()
+	require.NoError(t, err)
+
+	// Create a temporary file for use.
+	fh, err := ioutil.TempFile("", "nomad-autoscaler*.hcl")
+	require.NoError(t, err)
+	defer os.RemoveAll(fh.Name())
+
+	// Enabled by default if block is present.
+	cfg := &Agent{}
+	nomadSourceCfg := `source "nomad" {}`
+
+	_, err = fh.WriteString(nomadSourceCfg)
+	require.NoError(t, err)
+	require.NoError(t, parseFile(fh.Name(), cfg))
+
+	expected := []*PolicySource{
+		{
+			Name:       "nomad",
+			EnabledPtr: nil,
+			Enabled:    true,
+		},
+	}
+	assert.ElementsMatch(t, expected, cfg.PolicySources)
+
+	// Disable "nomad" policy source.
+	cfg = &Agent{}
+
+	noNomadSourceCfg := `
+source "nomad" {
+  enabled = false
+}`
+
+	// Reset the test file.
+	err = fh.Truncate(0)
+	require.NoError(t, err)
+	_, err = fh.Seek(0, 0)
+	require.NoError(t, err)
+
+	_, err = fh.WriteString(noNomadSourceCfg)
+	require.NoError(t, err)
+	require.NoError(t, parseFile(fh.Name(), cfg))
+
+	result := defaultConfig.Merge(cfg)
+	expected = []*PolicySource{
+		{
+			Name:       "file",
+			EnabledPtr: nil,
+			Enabled:    true,
+		},
+		{
+			Name:       "nomad",
+			EnabledPtr: ptr.BoolToPtr(false),
+			Enabled:    false,
+		},
+	}
+	assert.ElementsMatch(t, expected, result.PolicySources)
+
+	// Disable "file" policy source.
+	cfg = &Agent{}
+
+	noFileSourceCfg := `
+source "file" {
+  enabled = false
+}`
+
+	// Reset the test file.
+	err = fh.Truncate(0)
+	require.NoError(t, err)
+	_, err = fh.Seek(0, 0)
+	require.NoError(t, err)
+
+	_, err = fh.WriteString(noFileSourceCfg)
+	require.NoError(t, err)
+	require.NoError(t, parseFile(fh.Name(), cfg))
+
+	result = defaultConfig.Merge(cfg)
+	expected = []*PolicySource{
+		{
+			Name:       "file",
+			EnabledPtr: ptr.BoolToPtr(false),
+			Enabled:    false,
+		},
+		{
+			Name:       "nomad",
+			EnabledPtr: nil,
+			Enabled:    true,
+		},
+	}
+	assert.ElementsMatch(t, expected, result.PolicySources)
 }

--- a/agent/config/config_test.go
+++ b/agent/config/config_test.go
@@ -59,9 +59,8 @@ func TestAgent_Merge(t *testing.T) {
 		Policy: &Policy{
 			Sources: []*PolicySource{
 				{
-					Name:       "nomad",
-					EnabledPtr: ptr.BoolToPtr(false),
-					Enabled:    false,
+					Name:    "nomad",
+					Enabled: ptr.BoolToPtr(false),
 				},
 			},
 		},
@@ -112,14 +111,12 @@ func TestAgent_Merge(t *testing.T) {
 			DefaultEvaluationInterval: 10 * time.Second,
 			Sources: []*PolicySource{
 				{
-					Name:       "file",
-					EnabledPtr: ptr.BoolToPtr(false),
-					Enabled:    false,
+					Name:    "file",
+					Enabled: ptr.BoolToPtr(false),
 				},
 				{
-					Name:       "nomad",
-					EnabledPtr: ptr.BoolToPtr(true),
-					Enabled:    true,
+					Name:    "nomad",
+					Enabled: ptr.BoolToPtr(true),
 				},
 			},
 		},
@@ -214,14 +211,12 @@ func TestAgent_Merge(t *testing.T) {
 			DefaultEvaluationInterval: 10 * time.Second,
 			Sources: []*PolicySource{
 				{
-					Name:       "file",
-					EnabledPtr: ptr.BoolToPtr(false),
-					Enabled:    false,
+					Name:    "file",
+					Enabled: ptr.BoolToPtr(false),
 				},
 				{
-					Name:       "nomad",
-					EnabledPtr: ptr.BoolToPtr(true),
-					Enabled:    true,
+					Name:    "nomad",
+					Enabled: ptr.BoolToPtr(true),
 				},
 			},
 		},
@@ -455,9 +450,8 @@ policy {
 
 	expected := []*PolicySource{
 		{
-			Name:       "nomad",
-			EnabledPtr: nil,
-			Enabled:    true,
+			Name:    "nomad",
+			Enabled: ptr.BoolToPtr(true),
 		},
 	}
 	assert.ElementsMatch(t, expected, cfg.Policy.Sources)
@@ -485,14 +479,12 @@ policy {
 	result := defaultConfig.Merge(cfg)
 	expected = []*PolicySource{
 		{
-			Name:       "file",
-			EnabledPtr: nil,
-			Enabled:    true,
+			Name:    "file",
+			Enabled: ptr.BoolToPtr(true),
 		},
 		{
-			Name:       "nomad",
-			EnabledPtr: ptr.BoolToPtr(false),
-			Enabled:    false,
+			Name:    "nomad",
+			Enabled: ptr.BoolToPtr(false),
 		},
 	}
 	assert.ElementsMatch(t, expected, result.Policy.Sources)
@@ -520,14 +512,12 @@ policy {
 	result = defaultConfig.Merge(cfg)
 	expected = []*PolicySource{
 		{
-			Name:       "file",
-			EnabledPtr: ptr.BoolToPtr(false),
-			Enabled:    false,
+			Name:    "file",
+			Enabled: ptr.BoolToPtr(false),
 		},
 		{
-			Name:       "nomad",
-			EnabledPtr: nil,
-			Enabled:    true,
+			Name:    "nomad",
+			Enabled: ptr.BoolToPtr(true),
 		},
 	}
 	assert.ElementsMatch(t, expected, result.Policy.Sources)

--- a/command/agent.go
+++ b/command/agent.go
@@ -483,14 +483,14 @@ func (c *AgentCommand) readConfig() (*config.Agent, []string) {
 
 	if disableFileSource {
 		cmdConfig.Policy.Sources = append(cmdConfig.Policy.Sources, &config.PolicySource{
-			Name:       string(policy.SourceNameFile),
-			EnabledPtr: ptr.BoolToPtr(false),
+			Name:    string(policy.SourceNameFile),
+			Enabled: ptr.BoolToPtr(false),
 		})
 	}
 	if disableNomadSource {
 		cmdConfig.Policy.Sources = append(cmdConfig.Policy.Sources, &config.PolicySource{
-			Name:       string(policy.SourceNameNomad),
-			EnabledPtr: ptr.BoolToPtr(false),
+			Name:    string(policy.SourceNameNomad),
+			Enabled: ptr.BoolToPtr(false),
 		})
 	}
 

--- a/command/agent.go
+++ b/command/agent.go
@@ -345,10 +345,11 @@ func (c *AgentCommand) readConfig() (*config.Agent, []string) {
 		DynamicApplicationSizing: &config.DynamicApplicationSizing{},
 		HTTP:                     &config.HTTP{},
 		Nomad:                    &config.Nomad{},
-		Policy:                   &config.Policy{},
-		PolicyEval:               &config.PolicyEval{},
-		Telemetry:                &config.Telemetry{},
-		PolicySources:            []*config.PolicySource{},
+		Policy: &config.Policy{
+			Sources: []*config.PolicySource{},
+		},
+		PolicyEval: &config.PolicyEval{},
+		Telemetry:  &config.Telemetry{},
 	}
 
 	var disableFileSource bool
@@ -481,13 +482,13 @@ func (c *AgentCommand) readConfig() (*config.Agent, []string) {
 	}
 
 	if disableFileSource {
-		cmdConfig.PolicySources = append(cmdConfig.PolicySources, &config.PolicySource{
+		cmdConfig.Policy.Sources = append(cmdConfig.Policy.Sources, &config.PolicySource{
 			Name:       string(policy.SourceNameFile),
 			EnabledPtr: ptr.BoolToPtr(false),
 		})
 	}
 	if disableNomadSource {
-		cmdConfig.PolicySources = append(cmdConfig.PolicySources, &config.PolicySource{
+		cmdConfig.Policy.Sources = append(cmdConfig.Policy.Sources, &config.PolicySource{
 			Name:       string(policy.SourceNameNomad),
 			EnabledPtr: ptr.BoolToPtr(false),
 		})

--- a/command/agent.go
+++ b/command/agent.go
@@ -11,7 +11,9 @@ import (
 	"github.com/hashicorp/nomad-autoscaler/agent"
 	"github.com/hashicorp/nomad-autoscaler/agent/config"
 	agentHTTP "github.com/hashicorp/nomad-autoscaler/agent/http"
+	"github.com/hashicorp/nomad-autoscaler/policy"
 	flaghelper "github.com/hashicorp/nomad-autoscaler/sdk/helper/flag"
+	"github.com/hashicorp/nomad-autoscaler/sdk/helper/ptr"
 	"github.com/hashicorp/nomad-autoscaler/version"
 )
 
@@ -162,6 +164,14 @@ Policy Evaluation Options:
     <queue1>:<num>,<queue2>:<num>. Nomad Autoscaler supports "cluster" and
 	"horizontal" queues. Nomad Autoscaler Enterprise supports additional
 	"vertical_mem" and "vertical_cpu" queues.
+
+Policy Source Options:
+
+  -policy-source-disable-file
+    Disable the sourcing of policies from disk.
+
+  -policy-source-disable-nomad
+    Disable the sourcing of policies from the Nomad API.
 
 Telemetry Options:
 
@@ -338,7 +348,11 @@ func (c *AgentCommand) readConfig() (*config.Agent, []string) {
 		Policy:                   &config.Policy{},
 		PolicyEval:               &config.PolicyEval{},
 		Telemetry:                &config.Telemetry{},
+		PolicySources:            []*config.PolicySource{},
 	}
+
+	var disableFileSource bool
+	var disableNomadSource bool
 
 	modeChecker := config.NewModeChecker()
 
@@ -423,6 +437,10 @@ func (c *AgentCommand) readConfig() (*config.Agent, []string) {
 		return nil
 	}), "policy-eval-workers", "")
 
+	// Specify our Policy Sources flags.
+	flags.BoolVar(&disableFileSource, "policy-source-disable-file", false, "")
+	flags.BoolVar(&disableNomadSource, "policy-source-disable-nomad", false, "")
+
 	// Specify our Telemetry CLI flags.
 	flags.BoolVar(&cmdConfig.Telemetry.DisableHostname, "telemetry-disable-hostname", false, "")
 	flags.BoolVar(&cmdConfig.Telemetry.EnableHostnameLabel, "telemetry-enable-hostname-label", false, "")
@@ -460,6 +478,19 @@ func (c *AgentCommand) readConfig() (*config.Agent, []string) {
 	if err := modeChecker.ValidateFlags(flags); err != nil {
 		fmt.Printf("%s\n", err)
 		return nil, configPath
+	}
+
+	if disableFileSource {
+		cmdConfig.PolicySources = append(cmdConfig.PolicySources, &config.PolicySource{
+			Name:       string(policy.SourceNameFile),
+			EnabledPtr: ptr.BoolToPtr(false),
+		})
+	}
+	if disableNomadSource {
+		cmdConfig.PolicySources = append(cmdConfig.PolicySources, &config.PolicySource{
+			Name:       string(policy.SourceNameNomad),
+			EnabledPtr: ptr.BoolToPtr(false),
+		})
 	}
 
 	// Validate config values from flags.

--- a/command/agent_test.go
+++ b/command/agent_test.go
@@ -122,9 +122,11 @@ func TestCommandAgent_readConfig(t *testing.T) {
 				"-policy-source-disable-nomad",
 			},
 			want: defaultConfig.Merge(&config.Agent{
-				PolicySources: []*config.PolicySource{
-					{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
-					{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+				Policy: &config.Policy{
+					Sources: []*config.PolicySource{
+						{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+						{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+					},
 				},
 			}),
 		},
@@ -226,6 +228,10 @@ func TestCommandAgent_readConfig(t *testing.T) {
 					Dir:                       "./policy-dir-from-file",
 					DefaultCooldown:           12 * time.Second,
 					DefaultEvaluationInterval: 50 * time.Minute,
+					Sources: []*config.PolicySource{
+						{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+						{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+					},
 				},
 				PolicyEval: &config.PolicyEval{
 					DeliveryLimit:    10,
@@ -235,10 +241,6 @@ func TestCommandAgent_readConfig(t *testing.T) {
 						"cluster":    3,
 						"horizontal": 1,
 					},
-				},
-				PolicySources: []*config.PolicySource{
-					{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
-					{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
 				},
 			}),
 		},
@@ -318,6 +320,10 @@ func TestCommandAgent_readConfig(t *testing.T) {
 					Dir:                       "./policy-dir-from-file",
 					DefaultCooldown:           12 * time.Second,
 					DefaultEvaluationInterval: 50 * time.Minute,
+					Sources: []*config.PolicySource{
+						{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+						{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+					},
 				},
 				PolicyEval: &config.PolicyEval{
 					DeliveryLimit:    10,
@@ -327,10 +333,6 @@ func TestCommandAgent_readConfig(t *testing.T) {
 						"cluster":    3,
 						"horizontal": 1,
 					},
-				},
-				PolicySources: []*config.PolicySource{
-					{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
-					{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
 				},
 			}),
 		},

--- a/command/agent_test.go
+++ b/command/agent_test.go
@@ -116,6 +116,19 @@ func TestCommandAgent_readConfig(t *testing.T) {
 			}),
 		},
 		{
+			name: "policy source flags",
+			args: []string{
+				"-policy-source-disable-file",
+				"-policy-source-disable-nomad",
+			},
+			want: defaultConfig.Merge(&config.Agent{
+				PolicySources: []*config.PolicySource{
+					{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+					{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+				},
+			}),
+		},
+		{
 			name: "telemetry flags",
 			args: []string{
 				"-telemetry-disable-hostname",
@@ -223,6 +236,10 @@ func TestCommandAgent_readConfig(t *testing.T) {
 						"horizontal": 1,
 					},
 				},
+				PolicySources: []*config.PolicySource{
+					{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+					{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+				},
 			}),
 		},
 		{
@@ -311,6 +328,10 @@ func TestCommandAgent_readConfig(t *testing.T) {
 						"horizontal": 1,
 					},
 				},
+				PolicySources: []*config.PolicySource{
+					{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+					{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+				},
 			}),
 		},
 	}
@@ -321,11 +342,16 @@ func TestCommandAgent_readConfig(t *testing.T) {
 			got, _ := c.readConfig()
 
 			// Sort the list of plugins to avoid flakiness.
-			sortOpt := cmpopts.SortSlices(func(x, y *config.Plugin) bool {
+			pluginsSortOpt := cmpopts.SortSlices(func(x, y *config.Plugin) bool {
 				return x.Name < y.Name
 			})
 
-			if diff := cmp.Diff(tc.want, got, sortOpt); diff != "" {
+			// Sort the list of policy sources to avoid flakiness.
+			policySourcesSortOpt := cmpopts.SortSlices(func(x, y *config.PolicySource) bool {
+				return x.Name < y.Name
+			})
+
+			if diff := cmp.Diff(tc.want, got, pluginsSortOpt, policySourcesSortOpt); diff != "" {
 				t.Errorf("readConfig() mismatch (-want +got):\n%s", diff)
 			}
 		})

--- a/command/agent_test.go
+++ b/command/agent_test.go
@@ -124,8 +124,8 @@ func TestCommandAgent_readConfig(t *testing.T) {
 			want: defaultConfig.Merge(&config.Agent{
 				Policy: &config.Policy{
 					Sources: []*config.PolicySource{
-						{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
-						{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+						{Name: "file", Enabled: ptr.BoolToPtr(false)},
+						{Name: "nomad", Enabled: ptr.BoolToPtr(false)},
 					},
 				},
 			}),
@@ -229,8 +229,8 @@ func TestCommandAgent_readConfig(t *testing.T) {
 					DefaultCooldown:           12 * time.Second,
 					DefaultEvaluationInterval: 50 * time.Minute,
 					Sources: []*config.PolicySource{
-						{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
-						{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+						{Name: "file", Enabled: ptr.BoolToPtr(false)},
+						{Name: "nomad", Enabled: ptr.BoolToPtr(false)},
 					},
 				},
 				PolicyEval: &config.PolicyEval{
@@ -321,8 +321,8 @@ func TestCommandAgent_readConfig(t *testing.T) {
 					DefaultCooldown:           12 * time.Second,
 					DefaultEvaluationInterval: 50 * time.Minute,
 					Sources: []*config.PolicySource{
-						{Name: "file", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
-						{Name: "nomad", EnabledPtr: ptr.BoolToPtr(false), Enabled: false},
+						{Name: "file", Enabled: ptr.BoolToPtr(false)},
+						{Name: "nomad", Enabled: ptr.BoolToPtr(false)},
 					},
 				},
 				PolicyEval: &config.PolicyEval{

--- a/command/test-fixtures/agent_config_full.hcl
+++ b/command/test-fixtures/agent_config_full.hcl
@@ -40,6 +40,14 @@ policy {
   dir                         = "./policy-dir-from-file"
   default_cooldown            = "12s"
   default_evaluation_interval = "50m"
+
+  source "file" {
+    enabled = false
+  }
+
+  source "nomad" {
+    enabled = false
+  }
 }
 
 policy_eval {
@@ -50,12 +58,4 @@ policy_eval {
     cluster    = 3
     horizontal = 1
   }
-}
-
-source "file" {
-  enabled = false
-}
-
-source "nomad" {
-  enabled = false
 }

--- a/command/test-fixtures/agent_config_full.hcl
+++ b/command/test-fixtures/agent_config_full.hcl
@@ -23,21 +23,21 @@ nomad {
 }
 
 consul {
-  address = "https://consul_from_file.example.com:8500"
-  timeout = "2m"
-  token = "TOKEN_FROM_FILE"
-  auth = "user:file"
-  ssl =  true
+  address    = "https://consul_from_file.example.com:8500"
+  timeout    = "2m"
+  token      = "TOKEN_FROM_FILE"
+  auth       = "user:file"
+  ssl        = true
   verify_ssl = true
-  ca_file = "./ca-from-file.pem"
-  cert_file = "./cert-from-file.pem"
-  key_file = "./key-from-file.pem"
-  namespace = "namespace-from-file"
+  ca_file    = "./ca-from-file.pem"
+  cert_file  = "./cert-from-file.pem"
+  key_file   = "./key-from-file.pem"
+  namespace  = "namespace-from-file"
   datacenter = "datacenter-from-file"
 }
 
 policy {
-dir                         = "./policy-dir-from-file"
+  dir                         = "./policy-dir-from-file"
   default_cooldown            = "12s"
   default_evaluation_interval = "50m"
 }
@@ -50,4 +50,12 @@ policy_eval {
     cluster    = 3
     horizontal = 1
   }
+}
+
+source "file" {
+  enabled = false
+}
+
+source "nomad" {
+  enabled = false
 }

--- a/policy/manager.go
+++ b/policy/manager.go
@@ -62,6 +62,7 @@ func (m *Manager) Run(ctx context.Context, evalCh chan<- *sdk.ScalingEvaluation)
 
 	// Start the policy source and listen for changes in the list of policy IDs
 	for _, s := range m.policySource {
+		m.log.Info("starting policy source", "source", s.Name())
 		req := MonitorIDsReq{ErrCh: policyIDsErrCh, ResultCh: policyIDsCh}
 		go s.MonitorIDs(monitorCtx, req)
 	}

